ui/src/containers/UserProfile/index.tsx

```diff
@@ -15,16 +15,49 @@ import './UserProfile.css';
 const UserProfile: React.FC = () => {
   const { user } = useMst();
 
+  const [refreshId, setRefreshId] = useState<number>(0);
+  const [accessId, setAccessId] = useState<number>(0);
+
+  const triggerInterval = useCallback(() => {
+    const accessTokenInterval = user.accessTokenInfo.expiresAt * 1000 - new Date().getTime();
+    const refreshTokenInterval = user.refreshTokenInfo.expiresAt * 1000 - new Date().getTime();
+
+    // To get a new refresh token
+    // Update the refresh token before 10 seconds of current refresh token's expiry time
+    const tempRefreshId = window.setInterval(() => {
```

Comment:
I think we should create an action in store to handle this case or may be we can create a function in utils where we pass the tokens and get the result

Reply:
with action I had tried once but was not working